src/utils/cosmosdb_model.ts

```diff
@@ -264,6 +268,20 @@ export abstract class CosmosdbModel<
     );
   }
 
+  /**
+   * Add or update the ttl of the document identified by the searchKey
+   */
+
+  public updateTTL(
+    searchKey: DocumentSearchKey<TR, CosmosDocumentIdKey, PartitionKey>,
+    ttl: NonNegativeNumber
+  ): TE.TaskEither<CosmosErrors, TR> {
+    // this cast was necessary even making patch accept a Partial<TR>
+    return this.patch(searchKey, { ttl } as {
```

Comment:
The patch method should accept a `Partial<TR>` instead of a `Partial<T>`

Reply:
We tried it, and much more,  to make it compile, but it seems compiler doesn't like it. 
However we should solve it by passing the `ttl` in the BaseModel, as you suggested. We'll try.